### ISSUE TITLE
Hyperparameters to reproduce paper results

### ISSUE THREAD

**Issue by jeffra** [ipolharvard/ethos-paper#10]
Excellent work and great documentation in this repo, very much appreciated. My question relates to what set of hyperparameters should I use to reproduce your results in the paper.

I notice there are two different training scripts: 
1. Script in the usage section of the README https://github.com/ipolharvard/ethos-paper?tab=readme-ov-file#usage 
    *  uses n_layers=6 and dropout=0.3, but does not provide a LR or MIN_LR
2. https://github.com/ipolharvard/ethos-paper/blob/master/scripts/run_training.sh
    * uses n_layers=10, dropout=0.1, LR=0.0006, and MIN_LR=0.00001

After downloading the pre-trained model from google drive I see it is named `mimic_layer_6_batch_32_do_0.3` which implies this one uses n_layer=6 and dropout=0.3 which is great. If this is the right config, what was the LR/MIN_LR used for this model training?

**Comment by prenc** [comment]
Hi @jeffra,
Thank you for your kind words.

I am sorry for the confusion; the script for running training has not been properly updated, but be careful when using it, as it might not match your environment.

I double-checked, and the model that was used in the paper has the following parameters:
```bash
BATCH_SIZE=32
BLOCK_SIZE=2048
N_LAYER=6
N_HEAD=12
N_EMBD=768
DROPOUT=0.3
LR=0.0006
MIN_LR=0.00001
```

Also, if you want to train the model yourself, I suggest using the code from the `dev` branch with better token names and fixing some minor bugs.

**Comment by jeffra** [comment]
Excellent, thank you @prenc! I am currently training with this config on 8 x H100-80GB GPUs but on the `main` branch which should finish soon from the looks of it. After that completes I'll give the `dev` branch a try, but I guess I'd need to re-process the data from the looks of it?

**Comment by prenc** [comment]
That's correct, @jeffra. Unfortunately, the data has to be tokenized again to make sure the vocabulary is correct and compatible.
(Nice compute power!)

**Comment by jeffra** [comment]
Just pushed a PR to update the run script on the landing page to align with the specs you listed above
https://github.com/ipolharvard/ethos-paper/pull/11.
